src/Nixfmt/Types.hs

```diff
@@ -345,7 +345,7 @@ instance LanguageElement Expression where
           )
     (Assert _ cond _ body) -> [cond, body]
     (If _ expr0 _ expr1 _ expr2) -> [expr0, expr1, expr2]
-    (Abstraction param _ body) -> [Abstraction param (ann TColon) (Term (Token (ann (Identifier "_")))), body]
+    (Abstraction param _ body) -> walkSubprograms param ++ [body]
```

Comment:
I'd prefer a fix that does the reduction as initially proposed, and merely fixes the missing recursion anchor. Roughly:

```suggestion
    (Abstraction param _ (Term (Token (ann (Identifier "_")))) -> walkSubprograms param
    (Abstraction param _ body) -> [Abstraction param (ann TColon) (Term (Token (ann (Identifier "_")))), body]
```

Reply:
Ah I see, yeah that's better. I now changed it to:
https://github.com/NixOS/nixfmt/blob/4aaab5684d07ef8ea7ae75dd59fe005c034fb9cf/src/Nixfmt/Types.hs#L348-L353